src/Utilities/Testing.hpp

```diff
@@ -220,6 +220,35 @@ namespace GridKit
       return fail == 0;
     }
 
+    /**
+     * @brief Equatlity comparison between maps with a tolerance for the scalar value
+     *
+     * @tparam IdxT
+     * @tparam RealT
+     *
+     * @param[in] std::map<IdxT, RealT> - first map
+     * @param[in] std::map<IdxT, RealT> - second map
+     * @param[in] RealT - tolerance
+     * @return bool - true if the maps are equal; false otherwise
+     */
+    template <typename IdxT = size_t, typename RealT = double>
+    inline bool isEqual(std::map<IdxT, RealT> a,
+                        std::map<IdxT, RealT> b,
+                        const RealT           tol = std::numeric_limits<RealT>::epsilon())
+    {
+      int fail = 0;
+
+      fail += a.size() != b.size();
+
+      for (const auto& pair_a : a)
+      {
+        auto it  = b.find(pair_a.first);
+        fail    += !isEqual(pair_a.second, it->second);
```

Comment:
```suggestion
        fail    += !isEqual(pair_a.second, it->second, tol);
```

Should use tolerance here

Reply:
Also I think there should be some better error output here, like in the `std::vector` implementation - it would be good to output what pair we expected to find in `b` when we find something that is missing.